sigsci_site_manager/migrate.py

```diff
@@ -0,0 +1,164 @@
+import json
+import re
+from copy import deepcopy
+
+def get_dependency_items(item, found=None):
+    if found is None:
+        found = []
+    if isinstance(item, dict):
+        if any([x.startswith('corp.') for x in item.values() if isinstance(x, str)]):
+            if item not in found:
+                found.append(item)
+        for k, v in item.items():
+            if isinstance(v, (list, tuple, dict)):
+                found.extend([x for x in get_dependency_items(v, found=found) if x not in found])
+    elif isinstance(item, (list, tuple)):
+        for i in item:
+            found.extend([x for x in get_dependency_items(i, found=found) if x not in found])
+    return found
+
+def get_rule_dependencies(rule):
+    dependencies = {'signal': set(), 'rule_list': set()}
+    for item in get_dependency_items(rule):
+        try:
+            if item['operator'] == 'inList':
+                dependencies['rule_list'].add(item['value'])
+        except KeyError:
+            pass
+        try:
+            if item['type'] == 'addSignal':
+                dependencies['signal'].add(item['signal'])
+        except KeyError:
+            pass
+    return dependencies
+
+def get_corp_items(api):
+    corp_items = {}
+    corp_items['rule_list'] = {x['id']: x for x in 
+                                    api.get_corp_rule_lists()['data']}
+    corp_items['signal'] = {x['tagName']: x for x in 
+                                api.get_corp_signals()['data']}
+    return corp_items
+
+def get_alert_dependencies(alert):
+    if alert['tagName'].startswith('corp.'):
+        return alert['tagName']
+
+def get_advanced_rule_dependencies(advanced_rule):
+    dependencies = {'rule_list': set(), 'signal': set()}
+    matches = re.findall('[^\#]"(matchers|lists)*/*(corp\..+?)[")\s]',
+                   str(advanced_rule))
+    for match in matches:
+        if match[0] in ['matchers', 'lists']:
+            dependencies['rule_list'].add(match[1])
+        else:
+            dependencies['signal'].add(match[1])
+    return dependencies
+
+def format_dependencies(dependencies):
+    items = []
+    for item in dependencies:
+        if dependencies[item]:
+            items.append("'{}': {}".format(
+                item, ', '.join([str(x) for x in dependencies[item]])))
+    return "({})".format(', '.join(items))
+
+def migrate(api, file_name, output_file, dest_corp, strip, keep_users):
+    site_backup = json.load(open(file_name))
+    new_site = deepcopy(site_backup)
+    new_site['source']['corp'] = dest_corp
+    corp_dependencies = {'signal': set(), 'rule_list': set()}
+
+    # If not stripping dependencies, we need access to the same corp
+    if site_backup['source']['corp'] != api.corp and not strip:
+        raise Exception(
+            "CORP must be the same as backup file, otherwise use --strip.")
+    
+    msg = "Migrating '{}' from corp={} to corp={}.".format(
+        file_name, api.corp, dest_corp)
```

Comment:
Just as a general suggestion for the whole thing (I won't repeat this comment on all of them) I think `f""` styled format strings are easier to read because it's all inline.
```python
msg = f"Migrating '{file_name}' from corp={api.corp} to corp={dest_corp}."
```

Reply:
Fair. I can make that change.